numpyro/contrib/funsor/enum_messenger.py

```diff
@@ -442,10 +442,11 @@ def __init__(self, name, size, subsample_size=None, dim=None):
         if dim is not None and dim >= 0:
             raise ValueError('dim arg must be negative.')
         self.dim = dim
+        _, indices = OrigPlateMessenger._subsample(self.name, self.size, self.subsample_size, dim)
         self._indices = funsor.Tensor(
-            funsor.ops.new_arange(funsor.tensor.get_default_prototype(), self.size),
-            OrderedDict([(self.name, funsor.bint(self.size))]),
-            self.size
+            indices,
+            OrderedDict([(self.name, funsor.bint(self.subsample_size))]),
+            self.subsample_size
```

Comment:
@eb8680 I made those changes to support subsampling. Could you take a look to see if this is a reasonable change?

Reply:
Looks right to me. The corresponding code in `pyro.contrib.funsor` is here: https://github.com/pyro-ppl/pyro/blob/dev/pyro/contrib/funsor/handlers/plate_messenger.py#L63